SU-SWS/stanford_profile
# D8CORE-1472: Config Page for Lockup Configuration.

# READY FOR REVIEW

# Summary
- A configuration page for lockup options available to site managers

# Needed By (Date)
- Tuesday, July 28th

# Urgency
- Medium

# Steps to Test

1. Do a fresh build of the D8CORE-1472 branch of the acsf-cardinalsites stack with the stanford_profile (https://github.com/SU-SWS/acsf-cardinalsites/pull/176)
3. Clear caches
4. Go to /admin/config/system/lockup-settings
5. Select different lockup options from the drop down and see the available line input fields hide/show corresponding to the option selected.
6. Select an option other than `Logo Only` and fill out the line fields.
7. Save form and review on the home page you see what you would have expected to see.
8. Repeat with another option other than logo only.
9. Select `Logo Only` and save the form.
10. Review the home page and only see the Stanford wordmark.
11. Go back to the /admin/config/system/lockup-settings and uncheck `Use the logo supplied by the theme` so you can upload an image. 
12. Upload a large image.
13. Save the form and review the home page.
14. See only the uploaded logo and at a reasonable size.
15. Go back to /admin/config/system/lockup-settings and select `Option A` from the list of options and fill out the lines.
16. Save the form and go to the home page to see your uploaded logo and the lockup text. 

17. Create a new subtheme and install and enable it. You could use https://github.com/SU-SWS/stanford_starter
- OR -
17. Go to https://lockupsettings-dev.sites.stanford.edu/ and enable and set default the Victoria Sub Theme

18. Repeat logo/lockup changes for the subtheme to ensure it works on the active sub-theme. Styles may be off.


# TODO 
- Fix and update tests
- Code climate issues

# Affected Projects or Products
- Any profile with this config page added.

# Associated Issues and/or People
- https://stanfordits.atlassian.net/browse/D8CORE-1472
- https://github.com/SU-SWS/stanford_basic/pull/170
- @rmundstock 
- @buttonwillowsix 
- @katrialesser 
- @jenbreese  

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
